docs/evaluation/tutorials/eval_testing_tutorial.mdx

```diff
@@ -0,0 +1,813 @@
+    text_answer: Annotated[Optional[str], ..., "The text answer, if the user asked for one"]
+    reasoning: Annotated[str, ..., "The reasoning behind the answer"]
+
+tools = [code_tool, search_tool, ticker_tool]
+
+agent = create_react_agent(
+    model=ChatOpenAI(model="gpt-4o-mini"),
+    tools=tools,
+    response_format=AgentOutputFormat,
+    state_modifier="You are a financial expert. Respond to the users query accurately",
+)`,
+    },
+    {
+      value: "typescript",
+      label: "TypeScript",
+      language: "typescript",
+      content: `import { z } from "zod";
+import { ChatOpenAI } from "@langchain/openai";
+import { createReactAgent } from "@langchain/langgraph/prebuilt";
+
+const AgentOutputFormatSchema = z.object({
+    numeric_answer: z.number().optional().describe("The numeric answer, if the user asked for one"),
+    text_answer: z.string().optional().describe("The text answer, if the user asked for one"),
+    reasoning: z.string().describe("The reasoning behind the answer"),  
+})
+
+const tools = [codeTool, searchTool, tickerTool];
+
+const agent = createReactAgent({
+    llm: new ChatOpenAI({ model: "gpt-4o" }),
+    tools: tools,
+    responseFormat: AgentOutputFormatSchema,
+    stateModifier: "You are a financial expert. Respond to the users query accurately",
+}); 
+
+export default agent;`,
+    },
+  ]}
+  groupId="client-language"
+/>
+
+## Write tests
+
+Now that we have defined our agent, let's write a few tests to ensure basic functionality.
+In this tutorial we are going to test whether the agent's tool calling abilities are working, 
+whether the agent knows to ignore irrelevant questions, and whether it is able to answer complex questions
+that involve using all of the tools.
+
+### Setup test file
+
+Before we write any test, we need to set up our test file and add the imports needed at the top of the file.
+
+<CodeTabs
+  tabs={[
+    {
+      value: "python",
+      label: "Pytest",
+      language: "python",
+      caption: "Name your test file \`test_evals.py\`",
+      content: `from agent import agent # import from wherever your agent is defined
+import pytest
+import json
+from langsmith import testing`,
+    },
+    {
+      value: "vitest",
+      label: "Vitest",
+      language: "typescript",
+      caption: "Name your test file \`agent.vitest.eval.ts\`",
+      content: `import { expect } from "vitest";
+import * as ls from "langsmith/vitest";
+import agent from "../../src/agent"; // import from wherever your agent is defined`
+    },
+    {
+      value: "jest",
+      label: "Jest",
+      language: "typescript",
+      caption: "Name your test file \`agent.jest.eval.ts\`",
+      content: `import { expect } from "@jest/globals";
+import * as ls from "langsmith/jest";
+import agent from "../../src/agent"; // import from wherever your agent is defined`,
+    },
+  ]}
+  groupId="client-language"
+/>
+
+### Test 1: Ignore irrelevant questions
```

Comment:
It would be nice to have the complete file copy/pasteable

Maybe in a collapsed tab at the beginning/end?